### PR TITLE
Awstoken

### PR DESCRIPTION
**Pull request by mzanetti**

### CODE REVIEW

**Comment by mzanetti** [comment]
Jenkins, retest this please.

**Comment by mzanetti** [comment]
Jenkins, retest this please.

**Comment by mzanetti** [comment]
silo this

**Comment by mzanetti** [comment]
silo this

File: libguh-core/cloudmanager.cpp [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+void CloudManager::pairDevice(const QString &idToken, const QString &userId)
+{
+    m_awsConnector->pairDevice(idToken, userId);
+}
+
+void CloudManager::connect2aws()
+{
+    m_awsConnector->connect2AWS(m_serverUrl,
+//                                "1e10fb7e-d9d9-4145-88dd-2d3caf623c18",  // micha's test id (needs micha's test certs) - remove that before merging
```
Comment (t-mon) [inline libguh-core/cloudmanager.cpp]: Do what stands here :)

File: libguh-core/janusconnector.cpp [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+
+JanusConnector::JanusConnector(QObject *parent) : QObject(parent)
+{
+    m_socket = new QLocalSocket(this);
+    typedef void (QLocalSocket:: *errorSignal)(QLocalSocket::LocalSocketError);
+    connect(m_socket, static_cast<errorSignal>(&QLocalSocket::error), this, &JanusConnector::onError);
+    connect(m_socket, &QLocalSocket::disconnected, this, &JanusConnector::onDisconnected);
+    connect(m_socket, &QLocalSocket::readyRead, this, &JanusConnector::onReadyRead);
+    connect(m_socket, &QLocalSocket::stateChanged, [&](QLocalSocket::LocalSocketState state) {
+        qCWarning(dcJanus()) << "state changed" << state;
```
Comment (t-mon) [inline libguh-core/janusconnector.cpp]: This should maybe not be a warning

File: libguh-core/janusconnector.cpp [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+}
+
+void JanusConnector::onDisconnected()
+{
+    qCDebug(dcJanus) << "disconnected from Janus";
+}
+
+void JanusConnector::onError(QLocalSocket::LocalSocketError socketError)
+{
+    qCDebug(dcJanus) << "error in janus connection" << socketError << m_socket->errorString();
```
Comment (t-mon) [inline libguh-core/janusconnector.cpp]: This should be a warning.

File: libguh-core/jsonrpc/configurationhandler.cpp [hunk 4]
```diff
@@ -389,6 +402,21 @@ JsonReply *ConfigurationHandler::DeleteWebSocketServerConfiguration(const QVaria
     return createReply(statusToReply(GuhConfiguration::ConfigurationErrorNoError));
 }
 
+JsonReply *ConfigurationHandler::GetCloudEnabled(const QVariantMap &params) const
```
Comment (t-mon) [inline libguh-core/jsonrpc/configurationhandler.cpp]: maybe we should add this value to the basic configurations, which would save one call. What would not be a breaking change, just an extra value.

File: libguh-core/jsonrpc/jsonrpcserver.cpp [hunk 5]
```diff
@@ -140,6 +140,15 @@ JsonRPCServer::JsonRPCServer(const QSslConfiguration &sslConfiguration, QObject
     returns.insert("error", JsonTypes::userErrorRef());
     setReturns("RemoveToken", returns);
 
+    params.clear(); returns.clear();
```
Comment (t-mon) [inline libguh-core/jsonrpc/jsonrpcserver.cpp]: Is the JSONRPC namespace the right place for this method? I think this should go to Configurations or we should create a Cloud Namespace.

File: libguh-core/libguh-core.pro [hunk 6]
```diff
@@ -10,115 +10,125 @@ LIBS += -L$$top_builddir/libguh/ -lguh -lssl -lcrypto
 target.path = /usr/lib/$$system('dpkg-architecture -q DEB_HOST_MULTIARCH')
 INSTALLS += target
 
+exists("/usr/include/mbedtls/net_sockets.h") {
+DEFINES += MBEDTLS_NEW_HEADERS
```
Comment (t-mon) [inline libguh-core/libguh-core.pro]: a tab for the eye :)

**Comment by mzanetti** [comment]
Ooookay... lets see what the full ci stack thinks of this :)

Jenkins, land this please!

**Comment by mzanetti** [comment]
Jenkins, land this please

**Comment by mzanetti** [comment]
Jenkins, land this please

**Comment by mzanetti** [comment]
Jenkins, land this please

**Comment by t-mon** [comment]
Silo this.

**Comment by mzanetti** [comment]
silo this!

**Comment by mzanetti** [comment]
silo this

**Comment by mzanetti** [comment]
silo this

**Comment by mzanetti** [comment]
silo this dude!

**Comment by mzanetti** [comment]
Jenkins, land this please

**Comment by mzanetti** [comment]
Jenkins, land this please

**Pull request merged**